src/ast/ast_validations.c

```diff
@@ -1538,64 +1540,39 @@ static AST_Validation _extract_given_params(const cypher_astnode_t *statement,
 			const cypher_astnode_t *param = cypher_ast_cypher_option_get_param(option, j);
 			const char *paramName = cypher_ast_string_get_value(cypher_ast_cypher_option_param_get_name(param));
 			// If parameter already exists, add it the duplicated parms array.
-			if(!raxInsert(given_params_names, (unsigned char *) paramName, strlen(paramName), NULL, NULL))
-				duplicate_params = array_append(duplicate_params, paramName);
-		}
-	}
-	AST_Validation res = AST_VALID;
-	uint duplicate_params_count = array_len(duplicate_params);
-	if(duplicate_params_count > 0) {
-		asprintf(reason, "Duplicated parameters: %s", duplicate_params[0]);
-		for(uint i = 1; i < duplicate_params_count; i++) {
-			asprintf(reason, "%s, %s", *reason, duplicate_params[i]);
+			if(!raxInsert(given_params_names, (unsigned char *) paramName, strlen(paramName), NULL, NULL)) {
+				asprintf(reason, "Duplicated parameter: %s", paramName);
+				res = AST_INVALID;
```

Comment:
Missing `break` which will exit from both loops,
otherwise there's a potential leak here and a waste of cycles.

Reply:
replace with `return AST_INVALID`.